### ISSUE TITLE
`Style/MultipleComparison` autocorrect can cause syntax errors due to line breaks

### ISSUE THREAD

**Issue by corsonknowles** [rubocop/rubocop#13993]
## Expected behavior
Expected safe autocorrect to always return equivalent working code

## Actual behavior
In parenthesized multiline blocks, the correction causes syntax errors by splitting the `&&` operator to its own line. 

## Describe the solution you'd like

In this case, `&&` and similar operators must remain in the same line of code as the parenthetical expression that they belong to (i.e. stay with the correction).

## Describe alternatives you've considered

I have not been able to think of a more general solution

## Additional information

As an aside, autorcorrection here frequently leads to new violations of `Performance/CollectionLiteralInLoop` -- I know that could be a separate feature request, but I mention it here for completeness.

## Steps to reproduce the problem

Autocorrect this code:
```ruby
          (
            review.status == PerformanceManagement::V0::Review::Status::Draft ||
            review.status == PerformanceManagement::V0::Review::Status::NotStarted
          ) &&
            review.review_template_id == review_template_uuid
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.72.2 (using Parser 3.3.5.1, rubocop-ast 1.38.1, analyzing as Ruby 3.4, running on ruby 3.4.2) +server [arm64-darwin24]
  - rubocop-capybara 2.22.1
  - rubocop-factory_bot 2.27.1
  - rubocop-performance 1.24.0
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.5.0
  - rubocop-rspec_rails 2.31.0
  - rubocop-sorbet 0.9.0
  - rubocop-rspec 3.5.0
  - rubocop-rails 2.30.3
```

**Comment by Earlopain** [comment]
> As an aside, autorcorrection here frequently leads to new violations of Performance/CollectionLiteralInLoop -- I know that could be a separate feature request, but I mention it here for completeness.

Luckily such code is not beneficial anymore starting with Ruby 3.4 so in the future it should just go away https://github.com/rubocop/rubocop-performance/pull/488

**Comment by Earlopain** [comment]
This is an issue with `Style/RedundantParentheses`. Something like this triggers it (which you encounter through autocorrect):

```rb
(
  [foo,
   bar]
) &&
  baz
```

Where it then removes the parens, but, they are kinda necessary.

**Comment by Earlopain** [comment]
Looks like this was fixed with https://github.com/rubocop/rubocop/pull/14301 in RuboCop 1.77.0
